Add a simple import hook for pymetabiosis.

prabhuramachandran opened the pull request:
This hook first checks if the module is available in pypy and if not
found, it uses pymetabiosis to import the module in the embedded
CPython.

prabhuramachandran committed:
Add a simple import hook for pymetabiosis.

This hook first checks if the module is available in pypy and if not
found, it uses pymetabiosis to import the module in the embedded
CPython.

rguillebert commented on pymetabiosis/import_hook.py:
```diff
@@ -0,0 +1,59 @@
+    atexit.register(lambda: sys.meta_path.remove(import_hook))
+
+add_numpy_converters()
+install()
```
Please don't install it by default

prabhuramachandran replied:
Sure, I knew that would have to change.  I'll fix that.

prabhuramachandran replied:
Should I add a module called `auto.py` or something so people need only import that module instead of import followed by install?

prabhuramachandran committed:
Don't install import hook by default.

prabhuramachandran commented:
I've made the changes.  Could you also let me know about PR #14?

rguillebert commented:
Yeah, I'll review all the PRs this weekend

On Fri, Oct 2, 2015 at 1:12 PM, Prabhu Ramachandran <
notifications@github.com> wrote:

> I've made the changes. Could you also let me know about PR #14
> <https://github.com/rguillebert/pymetabiosis/pull/14>?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rguillebert/pymetabiosis/pull/20#issuecomment-145000531>
> .
>

prabhuramachandran commented:
Thanks!

rguillebert commented:
Tests? :)

prabhuramachandran commented:
@rguillebert -- yes, will add a few (I will have to think about how to do that though). I just wanted to check if you were OK with this in principle.  It seems this is fine, so I will work on this more and update the PR. Thanks for merging the other PRs. :)

prabhuramachandran committed:
Merge branch 'master' into import-hook

prabhuramachandran committed:
Add tests for import hook.

Also add a convenient auto.py module that automatically adds the hook on
import.

prabhuramachandran commented:
I've added tests, I also added a simple `auto.py` to make it easier for a user to just import a module and have the hook installed.  The travis-ci errors seem unrelated -- it is not able to install numpy.  It seems strange that though 2.6.1 is installed, the tests are using 2.5.1.

prabhuramachandran commented:
@rguillebert -- So the test failures are because of recent changes to numpy to use the out-of-line mode.  I am investigating this in #23.

rguillebert commented:
I've merged your other PR, try to rebase now :)

prabhuramachandran commented:
Thanks!  I will merge master into this branch, I prefer merges over rebase unless you would like me to rebase.

prabhuramachandran committed:
Merge branch 'master' of github.com:rguillebert/pymetabiosis into import-hook

prabhuramachandran commented:
Seems like all is well, please let me know if you want me to make any changes.

prabhuramachandran commented:
bump. :)

prabhuramachandran commented:
@rguillebert -- are there any issues you want me to address with this PR?  With this and #18, it makes it a lot easier to recommend pypy on non-Windows platforms as I can typically run many scripts with little or no modifications in pypy after the import hook is installed.

rguillebert commented:
I just need to review it one last time, I'll try to do this over the weekend

prabhuramachandran commented:
Thanks a ton!

prabhuramachandran commented:
@rguillebert -- gentle reminder! :)

prabhuramachandran commented:
@rguillebert -- is this project abandoned?
